plugins/modules/job_invocation.py

```diff
@@ -0,0 +1,223 @@
+        type: int
+extends_documentation_fragment:
+  - theforeman.foreman.foreman
+  - theforeman.foreman.foreman.entity_state_with_defaults
+  - theforeman.foreman.foreman.taxonomy
+'''
+
+EXAMPLES = '''
+
+- name: "Run remote command on a single host once"
+  job_invocation:
+    search_query: "name ^ (foreman.example.com)"
+    command: 'ls'
+    job_template_id: 148
+    ssh:
+      - effective_user: "tester"
+
+- name: "Run ansible commad on active hosts once a day"
+  job_invocation:
+    bookmark_id: 6
+    command: 'pwd'
+    job_template_id: 177
+    recurrence:
+      - cron_line: "30 2 * * *"
+    concurrency_control:
+      - concurrency_level: 2
+'''
+
+RETURN = ''' # '''
+
+from ansible_collections.theforeman.foreman.plugins.module_utils.foreman_helper import (
+    ForemanEntityAnsibleModule,
+)
+
+ssh_foreman_spec = {
+    'effective_user': dict(),
+}
+
+recurrnece_foreman_spec = {
+    'cron_line': dict(),
+    'max_iteration': dict(),
+    'end_time': dict(),
+}
+
+scheduling_foreman_spec = {
+    'start_at': dict(),
+    'start_before': dict(),
+}
+
+concurrency_control_foreman_spec = {
+    'time_span': dict(),
+    'concurrency_level': dict(),
+}
+
+
+class ForemanJobInvocationModule(ForemanEntityAnsibleModule):
+    pass
+
+
+def main():
+    module = ForemanJobInvocationModule(
+        foreman_spec=dict(
+            search_query=dict(),
+            bookmark_id=dict(),
+            job_template_id=dict(required=True),
+            targeting_type=dict(default='static_query', choices=['static_query', 'dynamic_query']),
+            randomized_ordering=dict(type='bool', default=False),
+            feature=dict(),
+            command=dict(),
+            inputs=dict(),
+            execution_timeout_interval=dict(),
+            ssh=dict(type='nested_list', foreman_spec=ssh_foreman_spec),
+            recurrence=dict(type='nested_list', foreman_spec=recurrnece_foreman_spec),
+            scheduling=dict(type='nested_list', foreman_spec=scheduling_foreman_spec),
+            concurrency_control=dict(type='nested_list', foreman_spec=concurrency_control_foreman_spec),
+        ),
+        entity_opts={'search_by': 'description'}
+    )
+
+    # make sure we have a search query
+    if 'search_query' not in module.foreman_params.keys() and 'bookmark_id' not in module.foreman_params.keys():
+        module.fail_json(
+            msg='No search query specified and no bookmark to infer it.')
+
+    # command type template validation
+    if module.foreman_params['job_template_id'] in ['148', '177'] and 'command' not in module.foreman_params.keys():
+        module.fail_json(
+            msg='No command specified for command template.')
+
+    # command input required by api
+    if 'command' in module.foreman_params.keys():
+        module.foreman_params['inputs'] = {"command": f"{module.foreman_params['command']}"}
+        module.foreman_params.pop('command')
+
+    with module.api_connection():
+        module.run()
```

Comment:
Because Bookmark lookup is not that easy, you may want to call something like
```
if 'bookmark' in module.foreman_params:
    module.set_entity('bookmark', module.find_resource('bookmark', search='name="{0}",controller="{1}"'.format(module.foreman_params['name'], module.foreman_params['controller']),
            failsafe=False,
        ))
```

Reply:
Is this rally required? Right now, bookmarks resolve just fine having `bookmark=dict(type='entity', controller=1)`

Reply:
Bookmarks are identified by the combination of their name and the controller.
It may work in your setup, because you have no duplicate names there atm.
I believe for this module, the controller will always be the host controller, but we cannot rely on it having the db-id 1.
Also `controller` is afaik not part of the `foreman_spec` specification, and i am wondering why it is simply ignored.